#
# This patch enables the usage of external CFLAGS
# (supplied via 'make CFLAGS="-O99'')
# without destroying the internal needed ones, by renaming the internals
#
--- procps-2.0.16-orig/Makefile	2003-09-15 23:49:34.000000000 +0200
+++ procps-2.0.16/Makefile	2004-09-03 14:43:01.000000000 +0200
@@ -1,20 +1,20 @@
 # Makefile for procps.  Chuck Blake.
 # Portions of this are highly dependent upon features specific to GNU make
 
-export PREFIX     =  #proc# prefix for program names
-export DESTDIR    =
-export MANDIR     =  /usr/share/man
-export MAN1DIR    =  $(DESTDIR)$(MANDIR)/man1
-export MAN5DIR    =  $(DESTDIR)$(MANDIR)/man5
-export MAN8DIR    =  $(DESTDIR)$(MANDIR)/man8
-export BINDIR     =  $(DESTDIR)/bin
-export SBINDIR    =  $(DESTDIR)/sbin
-export USRBINDIR  =  $(DESTDIR)/usr/bin
-export PROCDIR    =  $(DESTDIR)/usr/bin# /usr/proc/bin for Solaris devotees
-export OWNERGROUP =  --owner 0 --group 0
-export INSTALLBIN =  install
-export INSTALLLIB =  install
-export INSTALLMAN =  install --mode a=r
+export PREFIX     ?=  #proc# prefix for program names
+export DESTDIR    ?=
+export MANDIR     ?=  /usr/share/man
+export MAN1DIR    ?=  $(DESTDIR)$(MANDIR)/man1
+export MAN5DIR    ?=  $(DESTDIR)$(MANDIR)/man5
+export MAN8DIR    ?=  $(DESTDIR)$(MANDIR)/man8
+export BINDIR     ?=  $(DESTDIR)/bin
+export SBINDIR    ?=  $(DESTDIR)/sbin
+export USRBINDIR  ?=  $(DESTDIR)/usr/bin
+export PROCDIR    ?=  $(DESTDIR)/usr/bin# /usr/proc/bin for Solaris devotees
+export OWNERGROUP ?=  --owner 0 --group 0
+export INSTALLBIN ?=  install
+export INSTALLLIB ?=  install
+export INSTALLMAN ?=  install --mode a=r
 
 UPROG      =  uptime tload free w top vmstat watch skill snice pmap slabtop
 PPROG      =  pgrep pkill
@@ -27,17 +27,17 @@
 SUBDIRS    =  ps # sub-packages to build/install
 
 # easy to command-line override
-export GCC_WARN   =  -Wall -W -Wstrict-prototypes -Wshadow
-export INCDIRS    =  -I/usr/include/ncurses
-export CC         =  gcc
-export OPT        =  -O2
-export CFLAGS     =  -D_GNU_SOURCE $(OPT) $(GCC_WARN) -I$(shell pwd) $(INCDIRS)
-export SHARED     =  1# build/install both a static and ELF shared library
-export SHLIBDIR   =  $(DESTDIR)/lib
-export LDFLAGS    =  -Wl,-warn-common
-export SHLDFLAGS  =  $(LDFLAGS)
+export GCC_WARN   ?=  -Wall -W -Wstrict-prototypes -Wshadow
+export INCDIRS    ?=  -I/usr/include/ncurses
+export CC         ?=  gcc
+export OPT        ?=  -O2
+export CFLAGS     ?=  -D_GNU_SOURCE $(OPT) $(GCC_WARN) -I$(shell pwd) $(INCDIRS)
+export SHARED     ?=  1# build/install both a static and ELF shared library
+export SHLIBDIR   ?=  $(DESTDIR)/lib
+export LDFLAGS    ?=  -Wl,-warn-common
+export SHLDFLAGS  ?=  $(LDFLAGS)
 
-LIBCURSES  =  -lncurses# watch is the only thing that needs this
+LIBCURSES  ?=  -lncurses# watch is the only thing that needs this
 
 #----------------------------------------------------#
  End of user-configurable portion of the Makefile.  #
